app/src/main/java/org/simple/clinic/overdue/Appointment.kt

```diff
@@ -0,0 +1,82 @@
+package org.simple.clinic.overdue
+
+import android.arch.persistence.room.Dao
+import android.arch.persistence.room.Entity
+import android.arch.persistence.room.Insert
+import android.arch.persistence.room.OnConflictStrategy
+import android.arch.persistence.room.PrimaryKey
+import android.arch.persistence.room.Query
+import com.squareup.moshi.Json
+import io.reactivex.Flowable
+import org.simple.clinic.patient.SyncStatus
+import org.simple.clinic.util.RoomEnumTypeConverter
+import org.threeten.bp.Instant
+import org.threeten.bp.LocalDate
+import java.util.UUID
+
+@Entity(tableName = "Appointment")
+data class Appointment(
+    @PrimaryKey val id: UUID,
+    val patientId: UUID,
+    val facilityId: UUID,
+    val date: LocalDate,
+    val status: Status,
+    val statusReason: StatusReason,
```

Comment:
This should be `cancelReason` and not `statusReason`. This will also be set **ONLY** when an appointment is cancelled, so we will either have to
- Make it `nullable`, or
- Have a separate enum value which indicates no reason, which will be used when the appointment is in the `Scheduled` or `Visited` state.

Reply:
> This should be cancelReason and not statusReason

Backend seems to be using `statusReason` so I copied the same name.

> Have a separate enum value which indicates no reason, which will be used when the appointment is in the Scheduled or Visited state.

Right, so `NOT_CALLED_YET` is the default enum

Reply:
I am going by the model that was discussed on Slack. Since the usage of this field is actually a reason for cancellation, we should name it appropriately.

Reply:
Valid point. I wonder why did nilenso decide to change it to `status_reason`. I will ask them once they're back. I just want to keep our model namings in sync.